server/src/database/models/attainable.ts

```diff
@@ -10,24 +10,23 @@ import { sequelize } from '..';
 import Course from './course';
 import CourseInstance from './courseInstance';
 
-export default class CourseInstancePartialGrade extends Model<
-  InferAttributes<CourseInstancePartialGrade>,
-  InferCreationAttributes<CourseInstancePartialGrade>
+export default class Attainable extends Model<
+  InferAttributes<Attainable>,
+  InferCreationAttributes<Attainable>
 > {
   declare id: CreationOptional<number>;
   declare courseId: ForeignKey<Course['id']>;
   declare courseInstanceId: ForeignKey<CourseInstance['id']>;
-  declare type: string;
-  declare platform: string;
-  declare maxPoints: number;
-  declare minPoints: number;
-  declare weight: number;
-  declare expireAt: Date;
+  // TODO rename to parentId, atm sequelize forces name as "model + key" when querying.
+  declare attainableId: CreationOptional<ForeignKey<Attainable['id']>>;
+  declare name: string;
+  declare executionDate: Date; // Date when assignment is done (e.g., deadline or exam date)
```

Comment:
Perhaps `dueDate` then?

Reply:
Or we could also follow the same naming convention as sequelize and use `dueAt`. Reason why `executionDate` is used is because frontend also uses this and this way we have more harmonized environment. Any comments from @viivi-uhari ?

Reply:
I don't have string opinions about this but after thinking about this I think that `date` could actually be the best and most universal option. Since an exam isn't 'due' or assignments aren't executed in one day. Thoughts @no-comma ?

Reply:
`date` works if there is clear documentation about what it means.